### DISCUSSION TITLE
show loading component while waiting for page.server.tsx response

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#759] [Discussion]
Moved from Discord:

> hi @brillout, thank your for your project. I have a problem with long time process in .page.server.tsx. I want show loading component went waiting for page.server.tsx response, but i cannot found in document, please help me.
Thank you so much.

**Comment by brillout** [comment]
How about using `export { onPageTransitionStart, onPageTransitionEnd }`? See https://vite-plugin-ssr.com/clientRouting.

**Reply by Nelie-Taylor** [reply]
Let me try, thank you for support.

**Reply by brillout** [reply]
Thanks for sponsoring btw. 💚. Next step is to convince your employer 🙂.
